src/components/SessionActivityTracker.js

```diff
@@ -41,23 +36,32 @@ class SessionActivityTracker extends React.Component {
       this.setState({ counter: this.props.config.get('userSessionTimeoutInterval') })
     }
   }
+
   decrementCounter () {
-    if (this.state.counter !== null) {
-      const state = {
-        counter: this.state.counter - 1,
-      }
-      if (this.state.counter <= TIME_TO_DISPLAY_MODAL && !this.state.showTimeoutModal) {
-        state.showTimeoutModal = true
-      }
-      if (this.state.counter <= 0) {
+    const { counter } = this.state
+    if (counter !== null) {
+      this.setState({
```

Comment:
I think in this case it better to rewrite it like this 
```JavaScript
decrementCounter () {
  this.setState(state => {
    const { counter } = state
    if (counter !== null) {
      return {
        counter: counter - 1,
        showTimeoutModal: counter <= TIME_TO_DISPLAY_MODAL,
      }
    }
    return null
  }, () => {
    if (this.state.counter <= 0) {
      this.props.onLogout()
    }
  })
}
```

Reply:
I like that style but had to adjust the `appReady` condition in the `App` component to flip only after the config values have been loaded and `counter` won't default to `undefined`.